File: Add Code Here/C++/Longest_Substring_Without_Repeating_Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> charIndex(256, -1);
        int longest=0,m=0;
        for(int i=0;i<s.length();i++) {
            m=max(charIndex[s[i]]+1,m);
            charIndex[s[i]]=i;
            longest = max(longest,i-m+1);
        }
        return longest;
    }
};

File: Add Code Here/PYTHON/Longest Substring Without Repeating Characters.py
#s = "abcabcbb"
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        start = 0
        end = 0
        max_len = 0
        d = {}
        while end < len(s):
            if s[end] in d and d[s[end]] >= start:
                start = d[s[end]] + 1
            max_len = max(max_len, end - start + 1)
            d[s[end]] = end
            end += 1
        return max_len
